File: Baekjoon/Sorting/[2535]아시아 정보올림피아드/solution.py
N = int(input())

grade = []
m = {}

for i in range(N):
    nation, student, point = map(int, input().split())

    grade.append([nation, student, point])

grade.sort(key=lambda x: x[2], reverse=True)

cnt = 0
for n, s, p in grade:
    if(cnt == 3):
        break

    if (n in m) and m[n] == 2:
        continue

    if n in m:
        m[n] += 1
    else:
        m[n] = 1

    cnt += 1
    print(n, s)

File: Baekjoon/Sorting/[2535]아시아 정보올림피아드/solution.cpp
#include <algorithm>
#include <iostream>
#include <map>
#include <vector>

using namespace std;

#define endl "\n"

typedef pair<int, int> pii;

vector<pair<int, pii>> grade;
map<int, int> m;
int N;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);

  cin >> N;

  grade = vector<pair<int, pii>>(N, {0, {0, 0}});

  for (int i = 0; i < N; i++) {
    int nation, student, point;

    cin >> nation >> student >> point;

    grade[i] = {point, {nation, student}};
  }

  sort(grade.begin(), grade.end(), greater<>());

  int i, cnt;
  i = cnt = 0;
  while (cnt < 3) {
    if (m[grade[i].second.first] == 2) {
      i++;
      continue;
    }
    m[grade[i].second.first]++;
    cnt++;
    cout << grade[i].second.first << " " << grade[i].second.second << endl;
    i++;
  }

  return 0;
}
